llvm/utils/TableGen/DAGISelMatcherEmitter.cpp

```diff
@@ -88,6 +87,8 @@ class MatcherTableEmitter {
     DenseMap<const ComplexPattern *, unsigned> ComplexPatternUsage;
     // Record the usage of PatternPredicate.
     std::map<StringRef, unsigned> PatternPredicateUsage;
+    // Record the usage of Predicate.
+    DenseMap<TreePattern *, unsigned> PredicateUsage;
```

Comment:
This should use `MapVector`.  I've fixed `ComplexPatternUsage` in c230138011cbf07ad7caf9d256ae9d0c5032a974

Reply:
Thanks!